### PR TITLE
move brave/ads_service to BUILD.gn

### PR DESCRIPTION
**Pull request by bridiver**
Resolves https://github.com/brave/brave-browser/issues/41419

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/brave_news/BUILD.gn [hunk 1]
```diff
@@ -3,33 +3,6 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
-source_set("brave_news") {
```
Comment (bridiver) [inline browser/brave_news/BUILD.gn:6]: this target was not actually being used, these are also in sources.gni

**Review by aseren — APPROVED** [review]
LGTM, thanks!

File: browser/brave_ads/BUILD.gn [hunk 2]
```diff
@@ -3,6 +3,132 @@
...
+    "//brave/browser/ui/brave_ads",
+    "//brave/components/brave_ads/browser",
+    "//brave/components/brave_ads/browser/application_state",
+    "//brave/components/brave_ads/content/browser",
+    "//brave/components/brave_ads/core/public:headers",
+    "//brave/components/brave_rewards/common",
+    "//brave/components/brave_search/common",
+    "//brave/components/p3a_utils",
+    "//chrome/browser:browser_process",
+    "//chrome/browser:browser_public_dependencies",
```
Comment (bridiver) [inline browser/brave_ads/BUILD.gn:68]: This ensures that generated files from `//chrome/browser` run before this target. Any missing brave targets should go in `brave_chrome_browser_public_deps`

File: browser/sources.gni [hunk 3]
```diff
@@ -582,19 +581,17 @@ brave_chrome_browser_deps += brave_browser_web_discovery_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_prefs_deps
 
-brave_chrome_browser_allow_circular_includes_from = [ "//brave/browser/ui" ]
+brave_chrome_browser_allow_circular_includes_from = [
+  "//brave/browser/ui",
+  "//brave/browser/brave_ads:impl",
```
Comment (bridiver) [inline browser/sources.gni:586]: this allows `//brave/browser/brave_ads:impl` to use `//chrome/browser` dependencies without causing a gn check error

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by cdesouza-chromium — APPROVED** [review]
LGTM - much more straightforward.

**Review by fmarier — APPROVED** [review]
`browser/tor/test/BUILD.gn` :+1:

**Review by darkdh — APPROVED** [review]
++

**Pull request merged**